audtorch/transforms/transforms.py

```diff
@@ -487,6 +488,133 @@ def __repr__(self):
+        4
+
+    """
+    def __init__(self, coverage, max_width, value, axis):
+        self.coverage = coverage
+        self.max_width = max_width
+        self.value = value
+        self.axis = axis
+
+    def __call__(self, signal):
+
+        if isinstance(signal, np.ndarray):
+            signal = torch.from_numpy(signal)
+
+        signal_size = signal.shape[self.axis]
+
+        if signal_size < self.max_width:
+            raise RuntimeError(
+                'size along axis {} is smaller than `max_width`: '
+                '{} < {}'.format(self.axis, signal_size, self.max_width))
+
+        average_width = (self.max_width + 1) / 2
+        num_blocks = int(signal_size * self.coverage / average_width)
+        if self.coverage != 0.:
+            num_blocks = max(1, num_blocks)
+
+        return F.mask(signal, num_blocks, self.max_width, value=self.value,
+                      axis=self.axis)
+
+    def __repr__(self):
+        options = 'coverage={0}, max_width={1}, value={2}, axis={3}'.format(
+            self.coverage, self.max_width, self.value, self.axis)
+        return '{0}({1})'.format(self.__class__.__name__, options)
+
+
+class MaskSpectrogramTime(RandomMask):
+    r"""Randomly masks spectrogram along time axis.
+
+    See `RandomMask` for more details.
+
+    Note:
+        The time axis is derived from `Spectrogram`'s output shape.
+
+    Args:
+        coverage (float): proportion of signal to mask
+        max_width (int): maximum block size in number of samples. The default
+            value corresponds to a time span of 0.1 seconds of a signal
+            with `sr=16000` and stft-specifications of `window_size=320` and
+            `hop_size=160`. Default: `11`
+        value (float): mask value
+
+    Example:
+        >>> from librosa.display import specshow
+        >>> import matplotlib.pyplot as plt  # doctest: +SKIP
+        >>> a = torch.empty(65000).uniform_(-1, 1)
+        >>> t = Compose([Spectrogram(320, 160), MaskSpectrogramTime(0.1)])
+        >>> magnitude = t(a).squeeze().numpy()
+        >>> specshow(np.log10(np.abs(magnitude) + 1e-4)) # doctest: +SKIP
+        >>> plt.show()  # doctest: +SKIP
+
+    """
+    def __init__(self, coverage, *, max_width=11, value=0):
+        super().__init__(coverage, max_width, value, axis=1)
+
+
+class MaskSpectrogramFrequency(RandomMask):
+    r"""Randomly masks spectrogram along frequency axis.
+
+    See `RandomMask` for more details.
+
+    Note:
+        The frequency axis is derived from `Spectrogram`'s output shape.
+
+    Args:
+        coverage (float): proportion of signal to mask
+        max_width (int, optional): maximum block size in number of
+            frequency bins. The default value corresponds to approximately
+            5% of all frequency bins with stft-specifications of
+            `window_size=320` and `hop_size=160`. Default: `8`
+        value (float): mask value
+
+    Example:
+        >>> from librosa.display import specshow
```

Comment:
The same here:
I guess you have to add another # doctest: +SKIP here as this line seems to require matplotlib and fails in the automatic tests on travis.

Reply:
done